docs/source/integrations/triton.rst

```diff
@@ -1,3 +1,599 @@
+
+HTTP/REST APIs is disabled by default, though it can be enabled via ``BENTOML_USE_HTTP_TRITONSERVER`` environment variable.
+
+.. code-block:: bash
+
+    $ docker run --init --rm -p 3000:3000 triton-integration:cpu serve \
+                            --production --triton-options model-control-mode=explicit \
+                            --triton-options load-model=onnx_mnist --triton-options load-model=torchscript_yolov5s
+
+.. epigraph::
+    Currently, TritonRunner does not support running `Metrics server <https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/metrics.html>`_.
+    If you are interested in supporting the metrics server, please open an issue on :github:`GitHub <bentoml/BentoML/issues/new/choose>`
+
+Additionally, BentoML supervisors will allocate a random port for the gRPC server, hence ``grpc-port`` options that is passed to ``--triton-options`` will be omited.
+
+---------------
+
+Remarks
+~~~~~~~
+
+It is worth to mention that Triton Inference Server is a highly optimized model server that aims simplify multi-models serving.
+The following list of benefits and drawbacks are based on our experience building this integration:
+
+:raw-html:`<br />`
+
+The benefits Triton Inference Server brings:
+
+* **Performance enhancement**: Triton Inference Server's concurrent model execution [#concurrent_model_execution]_
+  enables multiple models to be executed concurrently on the same/multiple GPUs, which significantly improves the overall throughput.
+  There is no doubt that by simply imlemented in C++, Triton Inference Server triumphs over its Python counterparts in terms of performance and resource
+  utilization as it can take full advantage of the underlying hardware, instead of the limitations of Python's GIL [#constraints]_.
+
+* **Extensive configuration options**: To go along with its suite of features and optimizations, Triton Inference Server also provides
+  an extensive set of configuration options that can be used to customize models' behaviour, with ability to extend its capabilities.
+
+* **Framework supports and custom backends**: Triton supports all major frameworks, including Tensorflow, PyTorch, ONNX, TensorRT [#tensorrt_support]_ and OpenVINO,
+  which enables for a wide range of use-cases and empowers ML practitioners to do what they do best. Additionally, they also support custom backends
+  that can be used to implement custom inference logic.
+
+Among the powerful features Triton offers, it is worth to mention that Triton's design philosophy are engineering-driven, meaning users who excels in taking full advantages of Triton 
+are ML engineers who are familiar with the domain of MLOps. This is not to say that data scientists cannot use Triton, but rather that they will have to learn a bit more about MLOps
+and model serving to be able to fully utilize its features and capabilities.
```

Comment:
This paragraph may not be needed. It is answered below in the "Should I use Triton Inference Server" section.